synfig-studio/src/gui/workarea.cpp

```diff
@@ -631,41 +617,36 @@ WorkArea::load_meta_data()
 	if(data.size() && (data=="0" || data[0]=='f' || data[0]=='F'))
 		set_background_rendering(false);
 
-	data=canvas->get_meta_data("guide_x");
-	get_guide_list_x().clear();
+	//for the guide to be stored we have to store 2 floats x,y + 1 float angle in rad
+	data=canvas->get_meta_data("guide");
+	get_guide_list().clear();
 	while(!data.empty())
 	{
 		String::iterator iter(find(data.begin(),data.end(),' '));
-		String guide(data.begin(),iter);
-	    ChangeLocale change_locale(LC_NUMERIC, "C");
-
-		if(!guide.empty())
-			get_guide_list_x().push_back(stratof(guide));
-
-		if(iter==data.end())
-			data.clear();
-		else
-			data=String(iter+1,data.end());
-	}
-	//sort(get_guide_list_x());
-
-	data=canvas->get_meta_data("guide_y");
-	get_guide_list_y().clear();
-	while(!data.empty())
-	{
-		String::iterator iter(find(data.begin(),data.end(),' '));
-		String guide(data.begin(),iter);
-	    ChangeLocale change_locale(LC_NUMERIC, "C");
-
-		if(!guide.empty())
-			get_guide_list_y().push_back(stratof(guide));
+		String guide(data.begin(),iter);//this now contains the three items
+		std::vector<String> guide_components(3);
+		int i = 0;
+		for (auto character: guide){
+			if(character != '*')
+				guide_components.at(i).push_back(character);
+			else
+				++i;
+		}
+		if(!guide.empty()){
+			ChangeLocale change_locale(LC_NUMERIC, "C");
+			Guide obj = { synfig::Point{stratof(guide_components.at(0)),stratof(guide_components.at(1))},
+						  synfig::Angle::rad(stratof(guide_components.at(2)))
+						};
+			get_guide_list().push_back(obj);
+		} else{
+			synfig::warning("Error on trying to restore guide meta data");
```

Comment:
append: got %d components instead of 3

Reply:
I'm replying to the review before this. Checking `guide.size() != 3` wouldn't really make much sense guide is the string which contains the guide meta data which looks something like `guide_center_x_data*guide_center_y_data*guide_angle_data`. I think you confused it with `guide_components` which is the three component vector which we extract the data from guide string into it.

Reply:
done, but I am leaving this open in case you mean something else.

Reply:
> I think you confused it with `guide_components`

:+1: